src/fprime_gds/common/fpy/serialize_bytecode.py

```diff
@@ -0,0 +1,199 @@
+from __future__ import annotations
+from dataclasses import astuple
+import json
+from pathlib import Path
+from argparse import ArgumentParser
+import struct
+import zlib
+from fprime_gds.common.fpy.types import (
+    StatementTemplate,
+    StatementData,
+    Header,
+    Footer,
+    HEADER_FORMAT,
+    FOOTER_FORMAT,
+    StatementType,
+    directives
+)
+from fprime_gds.common.loaders.cmd_json_loader import CmdJsonLoader
+from fprime.common.models.serialize.array_type import ArrayType
+from fprime.common.models.serialize.bool_type import BoolType
+from fprime.common.models.serialize.enum_type import EnumType
+from fprime.common.models.serialize.numerical_types import (
+    F32Type,
+    F64Type,
+    I8Type,
+    I16Type,
+    I32Type,
+    I64Type,
+    U8Type,
+    U16Type,
+    U32Type,
+    U64Type,
+)
+from fprime.common.models.serialize.serializable_type import SerializableType
+from fprime.common.models.serialize.string_type import StringType
+from fprime.common.models.serialize.time_type import TimeBase, TimeType
+from fprime.common.models.serialize.type_base import BaseType, ValueType
+
+
+def get_type_obj_for(type: str) -> type[ValueType]:
+    if type == "FwOpcodeType":
+        return U32Type
+    elif type == "FwSizeStoreType":
+        return U16Type
+
+    raise RuntimeError("Unknown FPrime type alias " + str(type))
+
+
+def serialize_statement(stmt: StatementData) -> bytes:
+    # see https://github.com/nasa/fprime/issues/3023#issuecomment-2693051677
+    # TODO replace this with actual documentation
```

Comment:
The code is really easy to follow (thanks for that) so we don't need too much, but yes it'd be good to add a few docstrings to be able to understand what's going on without diving too deep.